python/pyspark/sql/tests/arrow/test_arrow_cogrouped_map.py

```diff
@@ -0,0 +1,301 @@
+#
```

Comment:
new test cases should be added into `dev/sparktestsupport/modules.py`. Also, we can do it separately but would be nicer if we add `test_parity_arrow_cogrouped_map.py` for Spark connect, e.g., see `pyspark.sql.tests.connect`